understory_transform_encoder/src/lib.rs

```diff
@@ -0,0 +1,2924 @@
+//! - [`InputValue1DBuilder::to_y()`] - Emit to Y axis only: `value → (0, value)`
+//! - [`InputValue1DBuilder::uniform()`] - Emit to both axes: `value → (value, value)`
+//! - [`InputValue1DBuilder::rotate()`] - Direct 1D→1D rotation: `value → rotate(value)`
+//!
+//! ### All Supported Mapping Combinations
+//!
+//! | Input → Output | Examples |
+//! |----------------|----------|
+//! | **2D → 2D**    | Drag → Pan, Drag → Non-uniform Scale |
+//! | **2D → 1D**    | Drag X → Rotate, Drag Magnitude → Rotate |
+//! | **1D → 2D**    | Scroll → Uniform Scale, Pinch → Pan Both Axes |
+//! | **1D → 1D**    | Rotate Gesture → Rotate Transform |
+//!
+//! ## Transform Actions: Low-Level Operations
+//!
+//! At the lowest level, the system uses [`TransformAction`] types to represent transform
+//! operations before backend-specific application:
+//!
+//! - [`TransformAction::Pan`] - Translation operations with [`PanAction`] variants
+//! - [`TransformAction::Scale`] - Scaling operations with [`ScaleAction`] variants  
+//! - [`TransformAction::Rotate`] - Rotation operations with [`RotateAction`] variants
+//!
+//! Each action type supports multiple operation modes:
+//! - **Fixed values**: Absolute positioning (`To` variants)
+//! - **Relative deltas**: Additive changes (`By`, `DeltaBy` variants)
+//! - **Multiplicative factors**: Scaling operations (`By` variants for scale/rotate)
+//!
+//! ## Transform Targets: Backend Implementations
+//!
+//! The [`TransformTarget`] trait allows different backends to handle transforms appropriately:
+//!
+//! - **[`Viewport1D`]** - 1D timeline/axis manipulation, maps 2D operations to 1D equivalents
+//! - **[`Viewport2D`]** - 2D canvas/map interaction with uniform scaling  
+//! - **[`kurbo::Affine`]** - Raw affine transforms supporting non-uniform scaling and rotation
+//!
+//! ### Direct Transform Usage
+//!
+//! ```rust
+//! use understory_transform_encoder::{TransformAction, TransformTarget, PanAction, ScaleAction};
+//! # #[cfg(feature = "view2d_adapter")]
+//! use understory_view2d::Viewport2D;
+//! use kurbo::{Rect, Vec2, Point};
+//!
+//! # #[cfg(feature = "view2d_adapter")]
+//! # {
+//! let mut viewport = Viewport2D::new(Rect::new(0.0, 0.0, 800.0, 600.0));
+//!
+//! // Apply transform actions directly
+//! viewport.apply(TransformAction::Pan(PanAction::By(Vec2::new(10.0, 5.0))));
+//! viewport.apply(TransformAction::Scale(ScaleAction::ByAbout {
+//!     scale: Vec2::new(1.5, 1.5),
+//!     anchor: Point::new(400.0, 300.0)
+//! }));
+//! # }
+//! ```
+
+extern crate alloc;
+
+use kurbo::{Affine, Point, Vec2};
+
+use ::ui_events::{
+    keyboard::{Key, KeyState, KeyboardEvent, Modifiers},
+    pointer::*,
+};
+use alloc::{rc::Rc, vec::Vec};
+
+/// Intent that describes a transform operation without specifying how to apply it.
+///
+/// Transform actions represent what the user wants to do (pan, scale, rotate) without
+/// being tied to a specific backend. The structure clearly maps to input types:
```

Comment:
I wouldn't say "clearly" here as that often means that it isn't actually clear :)